Route/RoutePageGenerator.php

```diff
@@ -99,9 +99,18 @@ public function update(SiteInterface $site, OutputInterface $output = null)
         }
 
         // Iterate over declared routes from the routing mechanism
+        $routesDedupInfos = array();
+        // We first de-duplicate routes by path; the first declaration win !
         foreach ($this->router->getRouteCollection()->all() as $name => $route) {
-            $name = trim($name);
-
+            $routePath = $route->getPath();
+            if (!array_key_exists($routePath, $routesDedupInfos)) {
```

Comment:
You could use avoid the logical negation by doing this : 

```php
if (array_key_exists($routePath, $routesDedupInfos)) {
    $this->writeln($output, sprintf(
        '  <info>IGNORE</info> route %s is ignored because it uses the same path as %s (path=%s)',
        $name,
        $routesDedupInfos[$routePath][0],
        $routePath
    ));
} else {
    $routesDedupInfos[$routePath] = array(trim($name), $route);
}
```